src/geos.c

```diff
@@ -880,3 +881,48 @@ GEOSGeometry* PyGEOSForce2D(GEOSContextHandle_t ctx, GEOSGeometry* geom) {
 GEOSGeometry* PyGEOSForce3D(GEOSContextHandle_t ctx, GEOSGeometry* geom, double z) {
   return force_dims(ctx, geom, 3, z);
 }
+
+GEOSCoordSequence* coordseq_from_buffer(GEOSContextHandle_t ctx, const double* buf,
+                                        unsigned int size, unsigned int dims, char ring_closure,
+                                        npy_intp cs1, npy_intp cs2) {
+  GEOSCoordSequence* coord_seq;
+  char *cp1, *cp2;
+  unsigned int i, j;
+  double first_coord;
+
+#if GEOS_SINCE_3_10_0
+
+  if ((!ring_closure) && (cs1 == dims * 8) && (cs2 == 8)) {
+    int hasZ = dims == 3;
+    coord_seq = GEOSCoordSeq_copyFromBuffer_r(ctx, buf, size, hasZ, 0);
+    return coord_seq;
+  }
```

Comment:
I think we could use `GEOSCoordSeq_copyFromArrays ` here if `(cs1 == 8) && (cs2 == size * 8)`

in that case `x = buf, y = buf + cs2, z = (dims == 3) ? buf + 2 * cs2 : NULL

Reply:
Looking into this, I am not sure it would actually work. At least not for "standard" fortran ordered arrays, because the strides are not exactly how we would need it:

```python
>>> coords = np.asarray(np.random.randn(10, 4, 2), order="F")
>>> coords.shape
(10, 4, 2)
>>> coords.strides
(8, 80, 320)
```

So the cs1 and cs2 is actually `total_size * 8` and `total_size * size * 8`, with `total_size` being 10 in this case (the total number of geometries that are going to be created), and that's something you don't know in the ufunc inner function + then it won't have the contiguous layout for CopyFromArrays anyway

So I could easily add the check you suggested, but I am not sure how useful it would be in practice. I need to do quite some axis wrangling in numpy to get an array with the correct strides to make use of this:

```
>>> coords = np.asarray(np.random.randn(4, 2, 10), order="F")
>>> coords = np.swapaxes(np.swapaxes(coords, 0, 2), 1, 2)
>>> coords.shape
(10, 4, 2)
>>> coords.strides
(64, 8, 32)
```

Reply:
I see. I guess it would 'work' for two-dimensional input arrays (scalar outputs), but in those cases you probably can't tell which GEOS function was used as the Python interpreter overhead would take up most of the time

Reply:
That's actually a good point. When creating a relatively large LineString or LinearRing (tested 1000 vertices), the difference is actually notable also in Python. So added that, and a test case using a scalar.